### PR TITLE
Invoice export XML can now have Firma ID as customer party identificator

### LINKED ISSUE #104: Validate invoice before being able to send to fakturabank
If you select an customer, and this customer have a scheme value and not a organisation nr.(accountplan.OrgNumber). The invoice will will not be ready to send to fakturabank. 
It will show  show `Mangler orgnummer`
it should show  a button with text `Fakturabank(F)`

### LINKED ISSUE #112: AccountPlan list page should show FirmaID instead of OrgNumber
Instead of OrgNumber in AccountPlan list page it should show Firma ID which will have the first in the list of Firma IDs for the AccountPlan formatted (for example NO:TELEPHONE 1234567).

### PR DESCRIPTION
**Pull request by mladjo2505**
Closes #104 
Also closes #112 (on same branch since here I created the getFirstFirmaID() method)

### CODE REVIEW

File: modules/fakturabank/model/fakturabank.class.php [hunk 1]
```diff
@@ -1875,16 +1875,14 @@ public function hash_to_xml($InvoiceO) {
...
+                  $cbc = $doc->createElement('cbc:CompanyID', utf8_encode($InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID));
+                  $cbc->setAttribute('schemeID', utf8_encode($InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyIDSchemeID));
+                  $legal_entity->appendChild($cbc);
 
-                        $cacparty->appendChild($legal_entity);
-                    } else {
-                        $_lib['message']->add("hash_to_xml::invalid orgnr. Organisasjonsnummeret " . $InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID . " til " . $InvoiceO->AccountingCustomerParty->Party->PartyName->Name . " er ugyldig. Fakturaen ble likevel sendt med kundenr som id.");
-                    }
+                  $cacparty->appendChild($legal_entity);
+                } else {
```
Comment (stoivo) [inline modules/fakturabank/model/fakturabank.class.php:1884]: Is't this `else` related to the iff you removed on line 1878?

Comment (stoivo) [inline modules/fakturabank/model/fakturabank.class.php:1884]: If it is, shouldn't we remove it?

File: modules/fakturabank/model/fakturabank.class.php [hunk 2]
```diff
@@ -1875,16 +1875,14 @@ public function hash_to_xml($InvoiceO) {
...
+                    $_lib['message']->add("hash_to_xml::invalid firma id. Firma ID " . $InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID . " til " . $InvoiceO->AccountingCustomerParty->Party->PartyName->Name . " er ugyldig. Fakturaen ble likevel sendt med kundenr som id.");
```
Comment (stoivo) [inline modules/fakturabank/model/fakturabank.class.php]: Treanslateded: (just so you know what this mean)

``` php
"hash_to_xml::invalid company id. Company ID " . $InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID . " to " . $InvoiceO->AccountingCustomerParty->Party->PartyName->Name . " in invalid. The invoice got sent anyway with customer no as id."
```

File: modules/fakturabank/model/fakturabank.class.php (hunk 1 above)
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php:1884]: No it is a fallback if the CompanyID node is empty, maybe just the message should be changed(?)
And possibly add a return statement so we do not send it?

File: modules/fakturabank/model/fakturabank.class.php (hunk 2 above)
Comment (mladjo2505) [inline modules/fakturabank/model/fakturabank.class.php]: @stoivo Maybe change this to a different message, and/or return to stop sending.

Comment (stoivo) [inline modules/fakturabank/model/fakturabank.class.php]: Can you check if we are validating `$InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID` anywhere earlier. If we already validate it, remove the else block and the conditional.

Change the message to "Firma ID til " . $InvoiceO->AccountingCustomerParty->Party->PartyName->Name . " er ugyldig. Fakturaen ble ikke sendt"
and return so we don't send.

**Pull request merged**